contracts/CPoR.sol

```diff
@@ -0,0 +1,139 @@
+pragma solidity ^0.5.16;
+
+import "./CErc20Delegate.sol";
+import "./AggregatorV3Interface.sol";
+
+/**
+ * @title Compound's CPoR (Proof of Reserves) Contract
+ * @notice CToken which checks reserves before minting
+ * @author Chainlink
+ */
+contract CPoR is CErc20Delegate, CPoRInterface {
+    /**
+     * @notice Construct a new money market
+     * @param underlying_ The address of the underlying asset
+     * @param comptroller_ The address of the Comptroller
+     * @param interestRateModel_ The address of the interest rate model
+     * @param initialExchangeRateMantissa_ The initial exchange rate, scaled by 1e18
+     * @param name_ ERC-20 name of this token
+     * @param symbol_ ERC-20 symbol of this token
+     * @param decimals_ ERC-20 decimal precision of this token
+     * @param admin_ Address of the administrator of this token
+     */
+    constructor(address underlying_,
+                ComptrollerInterface comptroller_,
+                InterestRateModel interestRateModel_,
+                uint initialExchangeRateMantissa_,
+                string memory name_,
+                string memory symbol_,
+                uint8 decimals_,
+                address payable admin_) public {
+        // Creator of the contract is admin during initialization
+        admin = msg.sender;
+
+        // Initialize the market
+        initialize(underlying_, comptroller_, interestRateModel_, initialExchangeRateMantissa_, name_, symbol_, decimals_);
+
+        // Set the proper admin now that initialization is done
+        admin = admin_;
+    }
+
+    /**
+     * @notice User supplies assets into the market and receives cTokens in exchange
+     * @dev Overrides CErc20's mintFresh function to check the proof of reserves
+     * @dev This check can be skipped if the feed is set to the zero address
+     * @param account The address of the account which is supplying the assets
+     * @param mintAmount The amount of the underlying asset to supply
+     * @return (uint, uint) An error code (0=success, otherwise a failure, see ErrorReporter.sol), and the actual mint amount.
+     */
+    function mintFresh(address account, uint mintAmount) internal returns (uint, uint) {
+        if (feed == address(0)) {
```

Comment:
If we read `feed` and `underlying` into memory variables at the beginning, we might be able to save some gas in this function. Given it's the hot path, the smallest savings will go a long way here.